types/uint.go

```diff
@@ -172,3 +180,43 @@ func checkNewUint(i *big.Int) (Uint, error) {
 	}
 	return Uint{i}, nil
 }
+
+
+// RelativePow raises x to the power of n, where x (and the result, z) are scaled by factor b
+// for example, RelativePow(210, 2, 100) = 441 (2.1^2 = 4.41)
+func RelativePow(x Uint, n Uint, b Uint) (z Uint) {
+	if x.IsZero() {
+		if n.IsZero() {
+			z = b // 0^0 = 1
+			return
+		}
+		z = ZeroUint() // otherwise 0^a = 0
+		return
+	}
+
+	if n.Mod(NewUint(2)).Equal(ZeroUint()) {
+		z = b
+	} else {
+		z = x
+
```

Comment:
```suggestion
```

Reply:
You can just avoid the else and do `z = x` as the default case and just if the single if statement.